src/typescript-worker/project-cache.ts

```diff
@@ -4,30 +4,41 @@
  * SPDX-License-Identifier: BSD-3-Clause
  */
 
-import {TypeScriptBuilder} from './typescript.js';
+import {LanguageServiceContext} from './language-service-context.js';
 import {CachingCdn} from './caching-cdn.js';
 import {ImportMapResolver} from './import-map-resolver.js';
 import {TypeScriptWorkerConfig} from '../shared/worker-api.js';
 
-export let builderInstance: TypeScriptBuilder;
+export let languageServiceContext: LanguageServiceContext;
 let cacheKey = '';
 
 /**
  * Acquire the existing builder instance, or create a fresh one if missing.
  * If the config differs from the existing instance's config, a new TypeScriptBuilder is
  * instantiated and made the new builderInstance.
+ *
+ * You can provide the caching cdn and moduleResolved instances yourself
+ * or let the function generate then from default values.
  */
-export function getBuilder(
+export function getLanguageServiceContext(
   config: TypeScriptWorkerConfig,
-  cdn: CachingCdn,
-  moduleResolver: ImportMapResolver
+  cdn?: CachingCdn,
```

Comment:
Since we're in a refactoring mood...

I think this function should *only* take a `config`. If there's a cache key hit, we already have a `CachingCdn` and `ImportMapResolver` in the stored context. Otherwise we create new ones from the config.